core/app/models/spree/order.rb

```diff
@@ -277,6 +277,13 @@ def create_tax_charge!
       Spree::TaxRate.adjust(self, shipments) if shipments.any?
     end
 
+    def update_line_item_prices!
+      line_items.each do |line_item|
+        line_item.update_price
+        line_item.save
```

Comment:
I wonder if we could improve this to save all line items at once to try squeezing performance gains here.  Quick google search isn't turning up much in the way of batching these records, but what if we do a single database transaction:
http://stackoverflow.com/a/2510258

Not sure how much better it would be, but I imagine it could add up for orders with many line items.